deltachat-jsonrpc/src/api/types/message.rs

```diff
@@ -173,6 +180,25 @@ impl MessageObject {
             Some(reactions.into())
         };
 
+        let vcard_contacts: Vec<VcardContact> = if view_type == Viewtype::Vcard {
+            if let Some(vcard_path) = message.get_file(context) {
+                if let Some(vcard) = tokio::fs::read(Path::new(&vcard_path))
+                    .await
+                    .ok()
+                    .as_ref()
+                    .and_then(|s| std::str::from_utf8(s).ok())
+                {
+                    parse_vcard(vcard).into_iter().map(Into::into).collect()
+                } else {
+                    Vec::new()
```

Comment:
Maybe just make `vcard_contacts` mutable? This 3-story vector initialisation looks too verbose